git_helper.go

```diff
@@ -0,0 +1,278 @@
+package publiccode
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"strings"
+)
+
+// Provides functionality for cloning and checking files in Git repositories.
+type gitHelper struct {
+	// Base directory for all temporary Git clones
+	tempDir string
+	// Maps repository URLs to their local clone paths
+	clonedRepos map[string]string
+}
+
+func newGitHelper() (*gitHelper, error) {
+	tempDir, err := os.MkdirTemp("", "publiccode-git-")
+	if err != nil {
+		return nil, fmt.Errorf("failed to create temp directory: %w", err)
+	}
+
+	return &gitHelper{
+		tempDir:     tempDir,
+		clonedRepos: make(map[string]string),
+	}, nil
+}
+
+// Performs a sparse clone of a Git repository.
+func (g *gitHelper) cloneRepo(repoURL string) (string, error) {
+	// Check if already cloned
+	if clonePath, ok := g.clonedRepos[repoURL]; ok {
+		return clonePath, nil
+	}
+
+	// Create a repo name
+	repoName := strings.NewReplacer(
+		"http://", "",
+		"https://", "",
+		"/", "_",
+		":", "_",
+	).Replace(repoURL)
+	if len(repoName) > 100 {
+		repoName = repoName[:100]
+	}
+
+	clonePath := filepath.Join(g.tempDir, repoName)
+
+	// Perform sparse clone
+	args := []string{"clone", "--filter=blob:none", "--no-checkout"}
+	args = append(args, repoURL, clonePath)
+	cmd := exec.Command("git", args...)
+
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return "", fmt.Errorf("git clone failed: %w\nOutput: %s", err, output)
+	}
+
+	// Initialize sparse checkout
+	cmd = exec.Command("git", "sparse-checkout", "init", "--cone")
```

Comment:
This creates a dependency on `git` and I'd rather not. Let's use a library here so it has a chance to work in WASM as well (if we implement storage for the clone someway)